apps/mobile/src/components/EncodedData/EncodedData.tsx

```diff
@@ -0,0 +1,23 @@
+import React, { useReducer } from 'react'
+import { Text } from 'tamagui'
+import { TouchableOpacity } from 'react-native'
+
+interface EncodedDataProps {
+  data: string
+}
+
+export function EncodedData({ data }: EncodedDataProps) {
+  const [truncated, toggleTruncate] = useReducer((state: boolean) => !state, true)
```

Comment:
just curious - why did you go for useReducer instead of useState?

Reply:
Since the state is a toggle, `useReducer` is a better approach for that.